APIAlfresco.php

```diff
@@ -1,475 +1,552 @@
-		}
-	}
-
-	/*
-	Create a folder inside the folder previously setted
-	$name = name of the folder to create.
-	@e.g.:
-		$name = "folder";
-	*/
-
-	public function createFolder($name,$properties = array(),$options = array()){
-		$exists = $this->existsFolder($name);
-		if ($exists){
-			print "Error:->The ".$name." folder already exists";
-			exit (255);
-		}
-		else{
-			return $this->repository->createFolder($this->parentFolder->id,$name);	
-		}
-	}
-
-	/*
-	
-	create file inside the folder previously setted
-	($this->parentFolder)
-	$name = name of the file to create.
-	@e.g.:
-		$name = "file.txt";
-		$content = "hi this is a file";
-	*/
-
-	public function createFile($name,$properties = array(),$content = null,$content_type = "application/octet-stream",$options = array()){
-		$exists = $this->FileExists($name);
-		if ($exists) {
-			print "Error:->the ".$name." file already exists";
-			//exit (255);
-		}
-		else{
-			return $this->repository->createDocument($this->parentFolder->id,$name,$properties,$content,$content_type,$options);
-		}
-	}
-
-	/*
-	upload files
-	@e.g.:
-		$file = "c:/temp/hello.pdf";
-	*/
-	
-	public function uploadFile($file){
-		
-		$name = basename($file);
-		$name = $this->extractSpecialCharacters($name);
-		$openFile = fopen($file, "r");
-		$content = fread($openFile, filesize($file));
-		//You need to activate fileinfo
-		$content_type = mime_content_type($file);
-		$newFile = $this->createFile($name,array(),$content,$content_type,array());
-		if ($newFile) {
-			return $newFile;
-		}
-			
-	}
-	/*
-	Download a file into temp directory and display the content by the browser
-	*/
-	
-	public function displayFile($id){
-		$file = $this->getObjectById($id);
-		$name = $file->properties["cmis:name"];
-		$mime = $file->properties["cmis:contentStreamMimeType"];
-		$mime = str_replace("\t", "", $mime);
-		$mime = str_replace("\n", "", $mime);
-		$length = $file->properties["cmis:contentStreamLength"];
-		$content = $this->repository->getContentStream($id);
-		$name = str_replace(" ", "_", $name);
-		$tempFile = fopen($name, "wb");
-		fwrite($tempFile, $content);
-		fclose($tempFile);
-		$domain = $_SERVER['SERVER_NAME'];
-		$path = getcwd()."/".$name;
-		header ("Cache-Control: must-revalidate, post-check=0, pre-check=0");
-        header('Content-type: '.$mime);        
+    /**
+     * Creates a file inside the folder previously setted ($this->parentFolder)
+     * $name = name of the file to create.
+     *
+     * @example $name = "file.txt";
+     * @example	$content = "hi this is a file";
+     *
+     * @param string $name
+     * @param string $content
+     * @param string $content_type
+     * @param array  $options
+     */
+    public function createFile($name, $properties = array(), $content = null, $content_type = 'application/octet-stream', $options = array())
+    {
+        $exists = $this->FileExists($name);
+        if ($exists) {
+            echo 'Error:->the '.$name.' file already exists';
```

Comment:
Throw exception or keep killing the process with exit(255), but this condition should maintain the consistency of previous functions.